packages/phone-number-privacy/combiner/src/salt-generation/get-threshold-salt.ts

```diff
@@ -48,63 +50,76 @@ export async function handleGetDistributedBlindedMessageForSalt(
       respondWithError(response, 401, ErrorMessages.UNAUTHENTICATED_USER)
       return
     }
-    const { successCount, signatures, majorityErrorCode } = await requestSignatures(request)
-    if (successCount >= config.thresholdSignature.threshold) {
-      const combinedSignature = await BLSCryptographyClient.combinePartialBlindedSignatures(
-        signatures,
-        request.body.blindedQueryPhoneNumber
-      )
-      response.json({ success: true, combinedSignature })
-    } else {
+    const { successCount, majorityErrorCode } = await requestSignatures(request, response)
+    if (successCount < config.thresholdSignature.threshold) {
       handleMissingSignatures(majorityErrorCode, response)
     }
   } catch (e) {
     respondWithError(response, 500, ErrorMessages.UNKNOWN_ERROR)
   }
 }
 
-async function requestSignatures(request: Request) {
+async function requestSignatures(request: Request, response: Response) {
   let successCount = 0
   const responses: SignMsgRespWithStatus[] = []
+  const signatures: ServicePartialSignature[] = []
   const errorCodes: Map<number, number> = new Map()
 
   const signers = JSON.parse(config.pgpnpServices.signers) as SignerService[]
-  const signerReqs = signers.map((service) =>
-    requestSigature(service, request)
-      .then(async (res) => {
+  const signerReqs = signers.map((service) => {
+    const controller = new AbortController()
+    const timeout = setTimeout(() => {
+      controller.abort()
+    }, config.pgpnpServices.timeoutMilliSeconds)
+    let sentResult: boolean = false
+
+    return requestSignature(service, request, controller)
+      .then(async (res: FetchResponse) => {
         const status = res.status
         if (res.ok) {
           const signResponse = (await res.json()) as SignMessageResponse
           responses.push({ url: service.url, signature: signResponse.signature, status })
+          signatures.push({ url: service.url, signature: signResponse.signature })
           successCount += 1
+          // Send response immediately once we cross threshold
+          if (!sentResult && successCount >= config.thresholdSignature.threshold) {
+            sentResult = true
+            const combinedSignature = await BLSCryptographyClient.combinePartialBlindedSignatures(
+              signatures,
+              request.body.blindedQueryPhoneNumber
+            )
+            response.json({ success: true, combinedSignature, version: VERSION })
```

Comment:
Can you confirm that this actually sends this response immediately instead of just adjusting the properties of the response object for express to handle later?

Reply:
Yeah, I've tested it out to confirm. While testing it I used @aslawson's method of running two signers: one normally and the other with a 1000s sleep. I've configured the new combiner timeout to be 100 seconds and verified that it returns the salt immediately after the non-slow signer has provided its signature. 

I originally found the approach via this article:
https://www.bennadel.com/blog/3275-you-can-continue-to-process-an-express-js-request-after-the-client-response-has-been-sent.htm